internal/command/version/update.go

```diff
@@ -46,5 +46,10 @@ func runUpdate(ctx context.Context) error {
 	}
 
 	io := iostreams.FromContext(ctx)
-	return update.UpgradeInPlace(ctx, io, release.Prerelease)
+
+	if err = update.UpgradeInPlace(ctx, io, release.Prerelease); err != nil {
+		return err
+	}
+	fmt.Fprintf(io.Out, "Updated flyctl v%s -> v%s\n", buildinfo.Version(), latest)
```

Comment:
I think this can show the wrong version for latest in some cases, like right after a release has happened. The reason is we query our web api to detect if there is a new version, and the web api sometimes has a slightly stale cached version.

I've seen cases like this:
```
$ fly version update 
Update available 0.0.450 -> v0.0.459.
...

$ fly version
fly v0.0.463 darwin/arm64 Commit: c44008d3 BuildDate: 2023-04-05T23:45:38Z
```

An alternate way to do this (and there are more!) would be run `fly version` in a subprocess so it would use the new binary. I didn't test this, but something along these lines might work:

```go
if output, err := exec.CommandContext(ctx, os.Args[0], "version").CombinedOutput(); err != nil {
    fmt.Fprintf(io.Out, "%s\n", output)
}
```

Reply:
I didn't end up calling into the new binary in this pr because I know windows can be... _fussy_ about binaries being moved, and I didn't have a windows machine handy to test on.

didn't think about the possibility there could be caching issues, though. thanks!